mono/metadata/marshal.c

```diff
@@ -1804,14 +1800,6 @@ emit_struct_conv_full (MonoMethodBuilder *mb, MonoClass *klass, gboolean to_obje
 		if (last_field) {
 			msize = klass->instance_size - info->fields [i].field->offset;
 			usize = info->native_size - info->fields [i].offset;
-
-			/* Make sure managed src pointer aligns with child struct address (if any) */
-			if(child_class_min_align)
-			{
-				msize += child_class_min_align - 1;
-				msize &= ~(child_class_min_align - 1);
-			}
-
```

Comment:
I don't think that this code was ever necessary. In the original test case added to the runtime tests, this code was not executed, since you need a base class that is not blittable to hit this case. I added one of those in my new runtime test, and this code causes it to fail.

I believe that since we are computing the managed size for each field due to field offsets, then we don't need to bother with alignment here, as the field offsets already account for alignment.